zzzz_modular_occulus/code/game/machinery/dorms_elevator.dm

```diff
@@ -0,0 +1,131 @@
+					to_chat(M, SPAN_NOTICE("Because you returned to the dorms, your crew respawn time has been reduced by [CRYOPOD_SPAWN_BONUS_DESC]."))
+					M << 'sound/effects/magic/blind.ogg' //Play this sound to a player whenever their respawn time gets reduced
+
+				//Going safely to cryo will allow the patient to respawn more quickly
+				M.set_respawn_bonus("CRYOSLEEP", CRYOPOD_SPAWN_BONUS)
+
+	// This despawn is not a gib() in this sense, it is used to remove objectives tied on these despawned mobs in cryos
+	passenger.ghostize()
+	passenger.despawn()
+
+
+
+//// This code is for the buttons and airlock in the elevator in the dorms area that offers players
+//// an alternative method of leaving the round without having to cryo.
+
+/obj/machinery/button/remote/elevator_panel
+	name = "Elevator Panel"
+	desc = "An elevator panel used to return to the dorms once a shift is over."
+
+	var/cooldown = 24 SECONDS
+	var/cooldown_timer
+	var/list/preserve_items = list(
+		/obj/item/weapon/hand_tele,
+		/obj/item/weapon/card/id/captains_spare,
+		/obj/item/device/aicard,
+		/obj/item/device/mmi,
+		/obj/item/device/paicard,
+		/obj/item/weapon/gun,
+		/obj/item/weapon/pinpointer,
+		/obj/item/clothing/suit,
+		/obj/item/clothing/shoes/magboots,
+		/obj/item/blueprints,
+		/obj/item/clothing/head/space,
+		/obj/item/weapon/storage/internal
+	)
+	var/obj/item/device/radio/intercom/announce
+	var/on_store_message = "has clocked out of their shift and returned to the dormitories."
+	var/on_store_name = "Elevator Announcer"
+
+/obj/machinery/button/remote/elevator_panel/New()
+	announce = new /obj/item/device/radio/intercom(src)
+	update_icon()
+	..()
+
+/obj/machinery/button/remote/elevator_panel/trigger()
+	var/elevator_prompt
+	var/turf/T = get_turf(src)
+
+	if(world.time - cooldown_timer > cooldown)
+		cooldown_timer = world.time
+		elevator_prompt = alert(src, "Warning! Using the elevator will end your shift and remove you from the round. Are you -sure- you want to head to your dorm for the day?\n(Everyone inside the elevator with you will end their shift and be removed as well!)", "End your shift and head to the dorms?", "End shift", "Stay on shift")
+		if(elevator_prompt == "End shift")
+			for(var/obj/machinery/door/airlock/D in SSmachines.machinery)
+				if (D.id_tag == id)
+					D.close()
+					sleep(11)
+					D.lock()
+			sleep(10)	//Wait 1 seconds before proceeding
+			playsound(src, 'zzzz_modular_occulus/sound/effects/elevator_going_up.ogg', 70, 0)
+			sleep(10)
+			visible_message("<span class='notice'>The elevator hums and creaks as it makes its way to the dormitory floor.</span>")
+			sleep(185)	//Wait 6 seconds before proceeding
+			for (var/mob/living/mobsgoingup in get_area(src))
+				despawn_passenger(mobsgoingup)
+		else
+			T.audible_message("<font color = #F63812><B>Elevator Coms</B>: \"Error. The elevator is currently out of service. Please contact your local engineering department.\"</font>")
```

Comment:
This would announce the error if the person interacting with the button decides to stay on shift.

Reply:
This is just a placeholder for now. Since the proc wasn't working to begin with, there was no way to say no, so I didn't bother moving it. It'll get changed once I get the PR fix out, along with all the other stuff as well